### PR TITLE
added guardians support

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added Guardians support:
- accept `options` field to be higher than `1`
- accept `guardian` field
- display `Guardian` field when signing guarded transactions
- fixed linter issues raised by the new libs
- added tests for the new scenarios

### CODE REVIEW

File: src/utils.c [hunk 1]
```diff
@@ -131,12 +131,13 @@ void append_to_str(const char* source, char* destination, int* index, int max_si
...
 
-void append_time(char* output, int* index, int max_size, int value, const char* tag) {
+void append_time(char* output, int* index, int value, const char* tag) {
     if (value == 0) {
         return;
     }
 
-    char buffer[max_size];
+    char buffer[MAX_INT_NUM_DIGITS + 1];
+    const int max_size = MAX_INT_NUM_DIGITS + 1;
```
Comment (sstanculeanu) [inline src/utils.c:140]:
```suggestion
    const int max_size = MAX_INT_NUM_DIGITS + 1;
    char buffer[max_size];
```

Comment (bogdan-rosianu) [reply src/utils.c:140]: not applicable. I've tries this but I get this linter issue:
```
src/utils.c:140:17: warning: variable length array used [-Wvla]
    char buffer[max_size];

```

**Review by sstanculeanu — DISMISSED** [review]

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: src/parse_tx.c [hunk 2]
```diff
@@ -298,10 +298,11 @@ static void extract_esdt_value(const char *encoded_data_field, const uint8_t enc
 // verify "chainID" field
 uint16_t verify_chainid(bool *valid) {
     if (strncmp(tx_hash_context.current_field, CHAINID_FIELD, strlen(CHAINID_FIELD)) == 0) {
-        char *ticker = TICKER_TESTNET;
+        char ticker[sizeof(TICKER_TESTNET)];
+        memmove(ticker, TICKER_TESTNET, strlen(TICKER_TESTNET) + 1);
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply src/parse_tx.c:302]: it was a linter issue. refactored

File: src/utils.c [hunk 3]
```diff
@@ -158,9 +159,8 @@ void seconds_to_time(const char* input, char* output, int max_size) {
         return;
     }
 
-    char temp_output[max_size];
+    char temp_output[MAX_AUTH_TOKEN_TTL_SIZE];
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply src/utils.c:162]: done

File: src/sign_tx_hash.c [hunk 4]
```diff
@@ -124,6 +124,11 @@ static void start_review(void) {
         pairs_list[step].item = "Value", pairs_list[step].value = tx_context.amount, ++step;
         pairs_list[step].item = "Receiver", pairs_list[step].value = tx_context.receiver, ++step;
         pairs_list[step].item = "Fee", pairs_list[step].value = tx_context.fee, ++step;
+        if (strlen(tx_context.guardian) > 0) {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply src/sign_tx_hash.c:127]: done

File: tests/test_multiversx.py [hunk 5]
```diff
@@ -371,6 +370,86 @@ def test_sign_tx_valid_simple_data_confirmed(self, backend, navigator, test_name
                            NavInsID.USE_CASE_REVIEW_CONFIRM]
                 navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name, nav_ins)
 
+    def test_sign_tx_valid_with_guardian(self, backend, navigator, test_name):
+        payload = b'{"nonce":1234,"value":"5678","receiver":"efgh","sender":"abcd","gasPrice":50000,"gasLimit":20,"chainID":"1","guardian":"erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th","version":2,"options":2,"data":"test"}'
+        with send_async_sign_message(backend, Ins.SIGN_TX_HASH, payload):
+            if backend.firmware.device.startswith("nano"):
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply tests/test_multiversx.py:376]: the `with` blocks are not the same. They have different expected messages and different number of actions before asserting something

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**